c7n/resources/glue.py

```diff
@@ -189,6 +190,85 @@ class resource_type(TypeInfo):
     augment = universal_augment
 
 
+@GlueCrawler.action_registry.register('update-crawler')
+class UpdateGlueCrawler(BaseAction):
+    """Updates glue crawlers based on specified parameter
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: glue-crawler-update
+                resource: aws.glue-crawler
+                filters:
+                  - type: value
+                    key: CrawlerSecurityConfiguration
+                    value: null
+                actions:
+                  - type: update-crawler
+                    attributes:
+                        CrawlerSecurityConfiguration: glue-approved
+
+    """
+
+    schema = type_schema(
+        'update-crawler',
+        required=['attributes'],
+        attributes={
+            'type': 'object',
+            'additionalProperties': False,
+            'properties': {
+                'Role': {'type': 'string'},
+                'DatabaseName': {'type': 'string'},
+                'Schedule': {'type': 'string'},
+                'Configuration': {'type': 'string'},
+                'CrawlerSecurityConfiguration': {'type': 'string'},
+                'TablePrefix': {'type': 'string'},
+                'Classifiers': {'type': 'array', 'items': {'type': 'string'}},
+                'SchemaChangePolicy': {
+                    'type': 'object',
+                    'additionalProperties': False,
+                    'properties': {
+                        'UpdateBehavior': {'type': 'string', 'enum': ['LOG', 'UPDATE_IN_DATABASE']},
+                        'DeleteBehavior': {
+                            'type': 'string', 'enum': [
+                                'LOG', 'UPDATE_IN_DATABASE', 'DEPRECATE_IN_DATABASE']}
+                    }
+                }
+            }
+        })
+
+    permissions = ('glue:UpdateCrawler',)
+    valid_origin_states = ('READY', 'FAILED')
+    crawler_mapping = {'Schedule': 'Schedule.ScheduleExpression'}
+
+    def process(self, crawlers):
+        crawlers = self.filter_resources(crawlers, 'State', self.valid_origin_states)
+        client = local_session(self.manager.session_factory).client('glue')
+        for crawler in crawlers:
+            self.process_resource(client, crawler)
+
+    def process_resource(self, client, crawler):
+        config = dict(self.data.get('attributes'))
+        modify_attrs = {}
+        for k, v in config.items():
+            if k == 'SchemaChangePolicy' and len(v) == 1:
+                for attr in ('UpdateBehavior', 'DeleteBehavior'):
+                    if attr not in v.keys():
+                        config[k][attr] = jmespath.search(k + '.' + attr, crawler)
```

Comment:
the jmespath stuff here seems like over kill for simple attribute lookups. this comparison feels like it could be simplified. also it would be nice to structure it as a separate method get_modifications / get_params / get_delta etc

Reply:
Gotcha, I'll simplify the implementation here! Thanks

Reply:
Got rid of complex jmespath key to use get() in get_delta(). Hope this looks simpler. Let me know if this one looks fine. Thanks!!